graphql_compiler/compiler/emit_sql.py

```diff
@@ -224,43 +243,72 @@ def _construct_fold_subquery(self, subquery_from_clause):
+        has_x_count = False
+
         # find outputs for this fold in all_folded_outputs and add to self._outputs
         if fold_scope_location.fold_path in all_folded_outputs:
             for fold_output in all_folded_outputs[fold_scope_location.fold_path]:
                 # distinguish between folds with the same fold path but different query paths
                 if (fold_output.base_location, fold_output.fold_path) == (
                         fold_scope_location.base_location, fold_scope_location.fold_path):
                     if fold_output.field == '_x_count':
-                        raise NotImplementedError(u'_x_count not implemented in SQL')
-
-                    # force column to have explicit label as opposed to anon_label
-                    intermediate_fold_output_name = 'fold_output_' + fold_output.field
-                    # add array aggregated output column to self._outputs
-                    self._outputs.append(
-                        sqlalchemy.func.array_agg(
-                            self.output_vertex_alias.c[fold_output.field]
-                        ).label(intermediate_fold_output_name)
-                    )
+                        has_x_count = True
+                        self._outputs.append(
+                            sqlalchemy.func.coalesce(
+                                sqlalchemy.func.count(),
+                                sqlalchemy.literal_column('0')
+                            ).label('fold_output_x_count')
+                        )
+
+                    else:
+                        # force column to have explicit label as opposed to anon_label
+                        intermediate_fold_output_name = 'fold_output_' + fold_output.field
+                        # add array aggregated output column to self._outputs
+                        self._outputs.append(
+                            sqlalchemy.func.array_agg(
+                                self.output_vertex_alias.c[fold_output.field]
+                            ).label(intermediate_fold_output_name)
+                        )
 
         # use to join unique identifier for the fold's outer vertex to the final table
         self._outputs.append(self.outer_vertex_alias.c[join_descriptor.from_column])
 
+        # _x_count must be added if _x_count is used in filtering, but is not an output
+        if not has_x_count:
+            for x_count_filter in x_count_filters:
+                if (x_count_filter.fold_scope_location.at_vertex() ==
+                        fold_scope_location.at_vertex()):
+                    self._outputs.append(
+                        sqlalchemy.func.coalesce(
+                            sqlalchemy.func.count(),
+                            sqlalchemy.literal_column('0')
+                        ).label('fold_output_x_count')
+                    )
+                    break
+
         return sorted(self._outputs, key=lambda column: column.name, reverse=True)
```

Comment:
Interesting, I'm very curious about this `reverse=True` :) I know it's not the subject of the PR though, so we can discuss it another time if that's easier.

Reply:
@jecdohmann any insight here?